### ISSUE TITLE
Overflow menu: Usage tab

### ISSUE THREAD

**Issue by thyhmdo** [carbon-design-system/carbon-website#3536]
**Ref Discussion** https://github.com/carbon-design-system/carbon-website/discussions/3132

## Usage tab

- [x] ~~**Question:** As per the discussion of "Is overflow menu a button variant?":
The Overflow menu component will be leveraging `menu` under the hood, as will things like context menu, combo button, and menu button. Overflow menu will be housed with `menu`instead of button.~~ This question has been resolved.

- [x] **Usage template:** Update content using the [usage template](https://carbondesignsystem.com/contributing/component/#usage-template:-for-components-with-one-variant).
- [x] **Formatting section:** This image should have a focus indicator on the first option in the menu. Maybe do a split image here of an overflow without the danger hover, showing the rule divider mentioned in the content above. Then show the overflow menu danger hover beneath.
<img width="899" alt="Screen Shot 2023-05-15 at 4 33 34 PM" src="https://github.com/carbon-design-system/carbon-website/assets/43969356/1eaceb3a-f098-47a3-9c45-fb0410c33256">

**Comment by laurenmrice** [comment]
Going to have a discussion and refine with Michael Gower.

**Comment by laurenmrice** [comment]
Blocked until we figure out the content structure of `Menu` in our website docs.

**Comment by laurenmrice** [comment]
Unblocked. We can move forward with placing the Overflow menu Usage/Style documentation within the new Menu buttons page on the website.

**Comment by laurenmrice** [comment]
This was resolved with #3700
